gpjax/types.py

```diff
@@ -31,6 +34,46 @@ def in_dim(self) -> int:
     def out_dim(self) -> int:
         return self.y.shape[1]
 
+    def _make_tfd_if_none(self) -> None:
+        if self._tf_dataset is None:
+            self._tf_dataset = tfd.Dataset.from_tensor_slices((self.X, self.y))
+    
+    def cache(self, *args, **kwargs) -> "Dataset":
```

Comment:
Can we just add a short, single sentence docstring to each of these methods please.